packages/router/src/grpc/custody/authorize/assert-valid-swaps.ts

```diff
@@ -0,0 +1,19 @@
+import { Address } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1/keys_pb';
+import { TransactionPlan } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/transaction/v1/transaction_pb';
+import { Code, ConnectError } from '@connectrpc/connect';
+
+export const assertValidSwaps = (
```

Comment:
when trying to swap the same pair, the extension throws "failed to deliver transaction: check_stateless failed: Trading pair must be distinct".

maybe we should have a separate `assertDistinctAssets` assertion check that handles that error with a dedicated toast?

Reply:
This is an interesting question. Should the extension only forcibly not allow harmful tx's or should we do some additional help? In this case, it feels like the outcome would be the same and that really the frontend needs to implement: https://github.com/penumbra-zone/web/issues/533

Reply:
Marking as resolved, as that's already captured in a separate ticket